### PR TITLE
Remove deprecated `kubectl rolling-update ...`

### PR DESCRIPTION
**Pull request by surajssd**
This commit removes the references to deprecated subcommand
`rolling-update`.

### CODE REVIEW

File: content/en/docs/reference/kubectl/cheatsheet.md [hunk 1]
```diff
@@ -163,19 +163,11 @@ kubectl get events --sort-by=.metadata.creationTimestamp
 
 ## Updating Resources
 
-As of version 1.11 `rolling-update` have been deprecated (see [CHANGELOG-1.11.md](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.11.md)), use `rollout` instead.
-
 ```bash
```
Comment (kbhawkey) [inline content/en/docs/reference/kubectl/cheatsheet.md]: Not sure on this change: May be helpful to keep the note about deprecating `rolling-update`?
Could add a single note, As of version 1.11 `kubectl rolling-update` has been deprecated (...).

Use `kubectl rollout` to <update a Deployment>?

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 2]
```diff
@@ -194,13 +194,11 @@ The detailed documentation of `kubectl autoscale` can be found [here](/docs/refe
...
-Horizontal Pod Autoscaler only supports the latter approach: the Horizontal Pod Autoscaler is bound to the deployment object,
-it sets the size for the deployment object, and the deployment is responsible for setting sizes of underlying replica sets.
+The Horizontal Pod Autoscaler is bound to the deployment object, it sets the size for
+the deployment object, and the deployment is responsible for setting sizes of underlying replica sets.
 
 Horizontal Pod Autoscaler does not work with rolling update using direct manipulation of replication controllers,
-i.e. you cannot bind a Horizontal Pod Autoscaler to a replication controller and do rolling update (e.g. using `kubectl rolling-update`).
+i.e. you cannot bind a Horizontal Pod Autoscaler to a replication controller and do rolling update
 The reason this doesn't work is that when rolling update creates a new replication controller,
 the Horizontal Pod Autoscaler will not be bound to the new replication controller.
```
Comment (kbhawkey) [inline content/en/docs/tasks/run-application/horizontal-pod-autoscale.md]: Is this complete?

File: content/en/docs/concepts/workloads/controllers/replicationcontroller.md [hunk 3]
```diff
@@ -202,19 +202,6 @@ As mentioned above, whether you have 1 pod you want to keep running, or 1000, a
...
-
-As explained in [#1353](http://issue.k8s.io/1353), the recommended approach is to create a new ReplicationController with 1 replica, scale the new (+1) and old (-1) controllers one by one, and then delete the old controller after it reaches 0 replicas. This predictably updates the set of pods regardless of unexpected failures.
-
-Ideally, the rolling update controller would take application readiness into account, and would ensure that a sufficient number of pods were productively serving at any given time.
-
-The two ReplicationControllers would need to create pods with at least one differentiating label, such as the image tag of the primary container of the pod, since it is typically image updates that motivate rolling updates.
-
-Rolling update is implemented in the client tool
-[`kubectl rolling-update`](/docs/reference/generated/kubectl/kubectl-commands#rolling-update). Visit [`kubectl rolling-update` task](/docs/tasks/run-application/rolling-update-replication-controller/) for more concrete examples.
-
```
Comment (kbhawkey) [inline content/en/docs/concepts/workloads/controllers/replicationcontroller.md]: Is `kubectl rolling-update` under the Deployment section on this page still relevant?

**Comment by Hartmarken348** [comment]
11920

**Comment by surajssd** [comment]
@janetkuo @kbhawkey @tfogo can you PTAL at the changes?

File: content/en/docs/concepts/workloads/controllers/replicationcontroller.md (hunk 3 above)
Comment (kbhawkey) [reply content/en/docs/concepts/workloads/controllers/replicationcontroller.md]: @surajssd, to fully clean-up, can you re-check the sections under the following Sub-headers on this page:

`Deployment`, remove `kubectl rolling-update`?
`Deleting just a ReplicationController`, there is a link that no longer works because you removed the content?

File: content/en/docs/reference/kubectl/cheatsheet.md (hunk 1 above)
Comment (kbhawkey) [reply content/en/docs/reference/kubectl/cheatsheet.md]: OK

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md (hunk 2 above)
Comment (kbhawkey) [reply content/en/docs/tasks/run-application/horizontal-pod-autoscale.md]: @surajssd, a couple of issues:
- With some of the content cut out, the remaining text loses some context for me.
- Some text is missing after `... and do rolling update The reason`

**Comment by kbhawkey** [comment]
Links to files changed by the PR:

https://deploy-preview-11920--kubernetes-io-master-staging.netlify.com/docs/tasks/run-application/horizontal-pod-autoscale/#autoscaling-during-rolling-update

https://deploy-preview-11920--kubernetes-io-master-staging.netlify.com/docs/reference/kubectl/cheatsheet

https://deploy-preview-11920--kubernetes-io-master-staging.netlify.com/docs/concepts/workloads/controllers/replicationcontroller/

https://deploy-preview-11920--kubernetes-io-master-staging.netlify.com/docs/concepts/cluster-administration/manage-deployment

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 4]
```diff
@@ -194,13 +194,11 @@ The detailed documentation of `kubectl autoscale` can be found [here](/docs/refe
 
 ## Autoscaling during rolling update
 
-Currently in Kubernetes, it is possible to perform a [rolling update](/docs/tasks/run-application/rolling-update-replication-controller/) by managing replication controllers directly,
-or by using the deployment object, which manages the underlying replica sets for you.
-Horizontal Pod Autoscaler only supports the latter approach: the Horizontal Pod Autoscaler is bound to the deployment object,
-it sets the size for the deployment object, and the deployment is responsible for setting sizes of underlying replica sets.
+The Horizontal Pod Autoscaler is bound to the deployment object, it sets the size for
```
Comment (janetkuo) [inline content/en/docs/tasks/run-application/horizontal-pod-autoscale.md]: HPA can target a Deployment, ReplicaSet, StatefulSet, and ReplicationController. 
It should be fixed in a separate PR, though. 
cc owners of this doc - @fgrzadkowski @jszczepkowski @directxman12

File: content/en/docs/tasks/run-application/horizontal-pod-autoscale.md [hunk 5]
```diff
@@ -194,13 +194,11 @@ The detailed documentation of `kubectl autoscale` can be found [here](/docs/refe
...
+the deployment object, and the deployment is responsible for setting sizes of underlying replica sets.
 
 Horizontal Pod Autoscaler does not work with rolling update using direct manipulation of replication controllers,
-i.e. you cannot bind a Horizontal Pod Autoscaler to a replication controller and do rolling update (e.g. using `kubectl rolling-update`).
+i.e. you cannot bind a Horizontal Pod Autoscaler to a replication controller and do rolling update
```
Comment (janetkuo) [inline content/en/docs/tasks/run-application/horizontal-pod-autoscale.md]: I think we can just kill this whole paragraph.

**Comment by Rajakavitha1** [comment]
@surajssd could you please resolve the new comments and suggestions.

**Comment by DanyC97** [comment]
@surajssd are you still interested in this PR ? if not please let us know so we can take it over

**Comment by DanyC97** [comment]
ping @surajssd , will be nice to get this changes in so we stop confusing users

**Comment by mistyhacks** [comment]
No comment from the OP since Jan 28. Feel free to reopen this PR and address feedback if you are still interested in getting this change into the docs.
